inspirehep/modules/workflows/tasks/arxiv.py

```diff
@@ -41,16 +41,14 @@
 from inspire_dojson.hep import hep
 from inspire_schemas.builders import LiteratureBuilder
 from inspire_schemas.utils import classify_field
-from inspirehep.modules.workflows.utils import convert
-from inspirehep.utils.record import get_arxiv_categories, get_arxiv_id
-from inspirehep.utils.url import is_pdf_link
-
 from plotextractor.api import process_tarball
 from plotextractor.converter import untar
 from plotextractor.errors import InvalidTarball, NoTexFilesFound
 
 from ..errors import DownloadError
-from ..utils import download_file_to_workflow, with_debug_logging
+from ..utils import download_file_to_workflow, with_debug_logging, convert
+from ....utils.record import get_arxiv_categories, get_arxiv_id
+from ....utils.url import is_pdf_link, retrieve_uri
```

Comment:
nit but...it's super hard to me reading relative imports :P

Reply:
([Google's Python style guide](https://google.github.io/styleguide/pyguide.html#Imports) actually disallows relative imports)

Reply:
Also here this is escaping the current module and reaching the `utils` folder, which is just confusing...

Reply:
Changed.

The fact that we have those "modules" (that actually are python subpackages, don't confuse with python modules) is by itself confusing. And imo a symptom of a wider structural issue in the general organization of the code.

Reply:
What I would like to have here is a fairly common architecture of large Flask applications based on blueprints, each implementing a distinct feature of INSPIRE: `editor`, `holdingpen`, `search`, `theme`...

This is what Explore Flask calls a "divisional structure": https://exploreflask.com/en/latest/blueprints.html#divisional. I detailed in several issues some of the steps needed to get there: https://github.com/inspirehep/inspire-next/issues/2251, https://github.com/inspirehep/inspire-next/issues/2260, https://github.com/inspirehep/inspire-next/issues/2641, https://github.com/inspirehep/inspire-next/issues/2642

Reply:
One thing that I see there is that it has, tops, a main package (top level) and supackages (one depth level) of python modules. That's something that's completely missed on all those issues.

Reply:
If what you want is to remove the `modules` folder I agree with that, but that's a much simpler (although quite wide) change than splitting code in their own blueprint-driven modules.

Mind you, we didn't even start with blueprints: in the past some modules were actually single-file Flask applications that were injecting themselves in the main application through internal Flask APIs...